instrumentation/logback/logback-appender-1.0/javaagent/src/test/groovy/LogbackTest.groovy

```diff
@@ -44,6 +44,13 @@ class LogbackTest extends AgentInstrumentationSpecification {
       waitForTraces(1)
     }
 
+    String jvmVersion = System.getProperty("java.vm.specification.version")
+    int codeAttributes = 3
+    boolean jvmVersionGreaterThanOrEqualTo18 = !jvmVersion.startsWith("1.8") && Integer.parseInt(jvmVersion) >= 18
+    if (jvmVersionGreaterThanOrEqualTo18) {
+      codeAttributes = 4 // Java 18 specificity on line number (lineNumber > 0 check)
```

Comment:
I wonder if this is because of groovy, I would be surprised to see different behavior for a regular Java class, I may try to convert this test to Java after merging it

Reply:
Line number is negative for Java <= 17, everything is fine for the library test based on JUnit 5

Reply:
Line number is negative for Java <= 17 and this Groovy test, everything is fine for the library test based on JUnit 5